configure.py

```diff
@@ -0,0 +1,63 @@
+import os, platform, subprocess
+
+def checkCommand(shell):
+    if platform.system() == "Windows":
+        return os.system("where " + shell)
+    else:
+        return os.system("which " + shell)
+
+def checkAll():
+    print("Checking for valid bash")
+    if checkCommand("bash"):
+        print("bash not found")
+        print("You will want a POSIX compatible environment")
+        exit(1)
+    else:
+        print("bash checked")
+    
+    print("Checking for rst2html")
+    if checkCommand("rst2html"):
+        print("rst2html not found")
+        print("Install with: pip3 install rst2html")
+        exit(1)
+    else:
+        print("rst2html checked")
+
+    print("Checking for python2")
+    if checkCommand("python2"):
+        print("python2 not found. ML still relies on it.")
+        exit(1)
+    else:
+        print("python2 works")
+
+    print("Checking for pip2")
+    if checkCommand("pip2"):
```

Comment:
My system has no pip2 and builds fine

Reply:
Don't the python2 scripts need some packages?

Reply:
Quite probably, but I don't have "pip2" so as is this check will fail my build environment.  If we need package X and Y available to python2, the appropriate check will be using python2 to import those packages.  This could be done as a one-liner per package, checking for success.

That would have the bonus of telling people what packages they need to install.

Reply:
So a python2 script that is called by python3?

Reply:
No, a direct check for the existence of the pre-requisite.  E.g., if we depend on the "os" module, this, which will succeed:
python3 -c 'import os'

Compare with a hypothetical "ossss" module, which will fail:
python3 -c 'import ossss'

It's not pip2 that we care about.  It's python2 (or 3), having access to certain modules.  So check for that directly.

Reply:
I was thinking of a try/catch usage to check if the imports failed.

Reply:
Sure, so was I, something like this:
try:
    subprocess.check_call(["python3", "-c", "import ossss"])
except CalledProcessError:
    # can't import ossss, handle this somehow

Reply:
I was thinking:
```
try:
    import ossss
except CalledProcessError:
    # can't import ossss, handle this somehow
```

Reply:
That feels a bit worse?  If you're running as python2 you can't test python3 imports, and vice versa.  My way any version of python can test imports for any other version.

Reply:
I guess so, for python2 a separate file would have to be made.

Reply:
Do you want to mark this one resolved?  The pip2 check isn't done anymore, but we're not checking for the modules.  Up to you.

Reply:
Sure